packages/elastic-apm-synthtrace/src/scripts/examples/04_span_links.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import uuid from 'uuid';
+import { apm, timerange } from '../../index';
+import { Scenario } from '../scenario';
+
+function generateExternalSpanLinks() {
+  // randomly creates external span links 0 - 10
+  return Array(Math.floor(Math.random() * 11))
+    .fill(0)
+    .map(() => ({ span: { id: uuid.v4() }, trance: { id: uuid() } }));
+}
+function generateEventsSpanLinks() {
+  const range = timerange(
+    new Date('2022-04-12T19:00:00.000Z'),
+    new Date('2022-04-12T19:05:00.000Z')
+  );
+  const instanceGo = apm.service('synth-go', 'production', 'go').instance('instance-a');
+  const events = range
+    .interval('1m')
+    .rate(1)
+    .spans((timestamp) => {
+      return instanceGo
+        .transaction('Sender')
+        .timestamp(timestamp)
+        .duration(1000)
+        .success()
+        .children(
+          instanceGo
+            .span('external', 'custom')
+            .timestamp(timestamp + 50)
+            .duration(100)
+            .success()
+        )
+        .serialize();
+    });
+
+  return events;
+}
+
+const scenario: Scenario = async () => {
+  return {
+    generate: ({ from, to }) => {
+      const externalSpanLinks = generateExternalSpanLinks();
+      const eventsSpanLinks = generateEventsSpanLinks();
```

Comment:
This sets up a generator that we're enumerating twice, once on line 54 and once when merging on 84.  

See this rebased example: 

[`main...Mpdreamz:apm-span-links-synthtrace?expand=1`#diff-8c951326a8](https://github.com/elastic/kibana/compare/main...Mpdreamz:apm-span-links-synthtrace?expand=1#diff-8c951326a8cc30ac0892fe27d90a93a7f3ef4d27c48a24fc1fbc721e5bee935e)

This will fix the id problem.

That said this is a usecase for scenario I had not yet anticipated nor one that previously existed. 

Scenario is setup to be able to return multiple streams but the assumptions is that they all share a same `to` and `from` which is not the case here. 

This messes up the OOTB range calculation to satisfy the `--maxDocs` parameter.


```
node src/scripts/run src/scripts/examples/04_span_links --local --maxDocs 100000 --clean
```